Frameworks/include/NSOrderedSetInternal.h

```diff
@@ -0,0 +1,28 @@
+//******************************************************************************
+//
+// Copyright (c) 2016 Microsoft Corporation. All rights reserved.
+//
+// This code is licensed under the MIT License (MIT).
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//******************************************************************************
+#pragma once
+
+#import <Starboard.h>
+#import <Foundation/NSOrderedSet.h>
+#import <map>
+#import <vector>
+
+@interface NSOrderedSet () {
+@protected
+    std::vector<StrongId<id>> setContainer;
+    std::map<id, std::vector<StrongId<id>>::iterator> mapContainer;
```

Comment:
Sorry for commenting super late here but what was the rational for using stl containers vs building on top of NSArray + NSSet? It seems like the general call patterns here may match closer with the NS counterparts and we can worry about optimizing / using a good backing store like stl there instead of re-implementing that "adapter" code. Just thinking that may be a bit easier in light of ben's concerns on iterators, the equals comparisons, etc...